// quran-meta/lib_cjs/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
const require_HafsLists = require('./lists/HafsLists.cjs');
const require_QalunLists = require('./lists/QalunLists.cjs');
const require_WarshLists = require('./lists/WarshLists.cjs');
const require_validation = require('./validation.cjs');
const require_getAyahCountInSurah = require('./getAyahCountInSurah.cjs');
const require_typeGuards = require('./typeGuards.cjs');
const require_ayahStringSplitter = require('./ayahStringSplitter.cjs');
const require_findAyahIdBySurah = require('./findAyahIdBySurah.cjs');
const require_findJuzByAyahId = require('./findJuzByAyahId.cjs');
const require_findJuz = require('./findJuz.cjs');
const require_findSurahAyahByAyahId = require('./findSurahAyahByAyahId.cjs');
const require_findSurahByAyahId = require('./findSurahByAyahId.cjs');
const require_findJuzAndShift = require('./findJuzAndShift.cjs');
const require_findJuzMetaBySurah = require('./findJuzMetaBySurah.cjs');
const require_findManzilByAyahId = require('./findManzilByAyahId.cjs');
const require_findManzil = require('./findManzil.cjs');
const require_findPage = require('./findPage.cjs');
const require_findPagebyAyahId = require('./findPagebyAyahId.cjs');
const require_findRukuByAyahId = require('./findRukuByAyahId.cjs');
const require_findRangeAroundAyah = require('./findRangeAroundAyah.cjs');
const require_findRangeAroundSurahAyah = require('./findRangeAroundSurahAyah.cjs');
const require_findRubAlHizbByAyahId = require('./findRubAlHizbByAyahId.cjs');
const require_findRubAlHizb = require('./findRubAlHizb.cjs');
const require_getList = require('./lists/getList.cjs');
const require_getRubAlHizb = require('./getRubAlHizb.cjs');
const require_getRubAlHizbByAyahId = require('./getRubAlHizbByAyahId.cjs');
const require_getAyahMeta = require('./getAyahMeta.cjs');
const require_getAyahMetasForSurah = require('./getAyahMetasForSurah.cjs');
const require_getJuzMeta = require('./getJuzMeta.cjs');
const require_getManzilMeta = require('./getManzilMeta.cjs');
const require_getPageMeta = require('./getPageMeta.cjs');
const require_getRubAlHizbMeta = require('./getRubAlHizbMeta.cjs');
const require_getRubAlHizbMetaByAyahId = require('./getRubAlHizbMetaByAyahId.cjs');
const require_getRukuMeta = require('./getRukuMeta.cjs');
const require_getSurahMeta = require('./getSurahMeta.cjs');
const require_prevAyah = require('./prevAyah.cjs');
const require_nextAyah = require('./nextAyah.cjs');
const require_isAyahJuzFirst = require('./isAyahJuzFirst.cjs');
const require_isAyahPageFirst = require('./isAyahPageFirst.cjs');
const require_isSurahAyahJuzFirst = require('./isSurahAyahJuzFirst.cjs');
const require_isSurahAyahPageFirst = require('./isSurahAyahPageFirst.cjs');
const require_surahStringParser = require('./surahStringParser.cjs');
const require_QuranRiwaya = require('./QuranRiwaya.cjs');
const require_index = require('./lists/index.cjs');
const require_surah_en = require('./i18n/surah.en.cjs');
const require_surah_az = require('./i18n/surah.az.cjs');
const require_surah_ru = require('./i18n/surah.ru.cjs');
const require_surah_tr = require('./i18n/surah.tr.cjs');
const require_surah_uz = require('./i18n/surah.uz.cjs');
const require_surah_kk = require('./i18n/surah.kk.cjs');
const require_surah_bs = require('./i18n/surah.bs.cjs');
const require_surah_fr = require('./i18n/surah.fr.cjs');
const require_surah_lt = require('./i18n/surah.lt.cjs');
const require_surah_ky = require('./i18n/surah.ky.cjs');
const require_surah_tg = require('./i18n/surah.tg.cjs');
const require_getSurahNames = require('./i18n/getSurahNames.cjs');
const require_types = require('./i18n/types.cjs');
const require_index$1 = require('./i18n/index.cjs');

//#region src/index.ts
/**
* Create a QuranRiwaya instance with Hafs riwaya (default)
*/
const createHafs = () => require_QuranRiwaya.QuranRiwaya.create(require_HafsLists.HafsLists);
/**
* Create a QuranRiwaya instance with Qalun riwaya
*/
const createQalun = () => require_QuranRiwaya.QuranRiwaya.create(require_QalunLists.QalunLists);
/**
* Create a QuranRiwaya instance with Warsh riwaya
*/
const createWarsh = () => require_QuranRiwaya.QuranRiwaya.create(require_WarshLists.WarshLists);
/**
* Pre-initialized QuranRiwaya instance for Hafs (for convenience)
*/
const quran = createHafs();

//#endregion
exports.QuranRiwaya = require_QuranRiwaya.QuranRiwaya;
exports.ayahStringSplitter = require_ayahStringSplitter.ayahStringSplitter;
exports.checkValidAyahId = require_validation.checkValidAyahId;
exports.checkValidJuz = require_validation.checkValidJuz;
exports.checkValidManzil = require_validation.checkValidManzil;
exports.checkValidPage = require_validation.checkValidPage;
exports.checkValidRuku = require_validation.checkValidRuku;
exports.checkValidSurah = require_validation.checkValidSurah;
exports.checkValidSurahAyah = require_validation.checkValidSurahAyah;
exports.createHafs = createHafs;
exports.createQalun = createQalun;
exports.createWarsh = createWarsh;
exports.findAyahIdBySurah = require_findAyahIdBySurah.findAyahIdBySurah;
exports.findJuz = require_findJuz.findJuz;
exports.findJuzAndShift = require_findJuzAndShift.findJuzAndShift;
exports.findJuzAndShiftByAyahId = require_findJuzAndShift.findJuzAndShiftByAyahId;
exports.findJuzByAyahId = require_findJuzByAyahId.findJuzByAyahId;
exports.findJuzMetaBySurah = require_findJuzMetaBySurah.findJuzMetaBySurah;
exports.findManzil = require_findManzil.findManzil;
exports.findManzilByAyahId = require_findManzilByAyahId.findManzilByAyahId;
exports.findPage = require_findPage.findPage;
exports.findPagebyAyahId = require_findPagebyAyahId.findPagebyAyahId;
exports.findRangeAroundAyah = require_findRangeAroundAyah.findRangeAroundAyah;
exports.findRangeAroundSurahAyah = require_findRangeAroundSurahAyah.findRangeAroundSurahAyah;
exports.findRubAlHizb = require_findRubAlHizb.findRubAlHizb;
exports.findRubAlHizbByAyahId = require_findRubAlHizbByAyahId.findRubAlHizbByAyahId;
exports.findRukuByAyahId = require_findRukuByAyahId.findRukuByAyahId;
exports.findSurahAyahByAyahId = require_findSurahAyahByAyahId.findSurahAyahByAyahId;
exports.findSurahByAyahId = require_findSurahByAyahId.findSurahByAyahId;
exports.generatePartBlocks = require_getList.generatePartBlocks;
exports.getAyahCountInSurah = require_getAyahCountInSurah.getAyahCountInSurah;
exports.getAyahMeta = require_getAyahMeta.getAyahMeta;
exports.getAyahMetasForSurah = require_getAyahMetasForSurah.getAyahMetasForSurah;
exports.getJuzMeta = require_getJuzMeta.getJuzMeta;
exports.getList = require_getList.getList;
exports.getListNormalised = require_getList.getListNormalised;
exports.getListOfRiwaya = require_index.getListOfRiwaya;
exports.getListsOfRiwaya = require_index.getListsOfRiwaya;
exports.getManzilMeta = require_getManzilMeta.getManzilMeta;
exports.getPageMeta = require_getPageMeta.getPageMeta;
exports.getRubAlHizb = require_getRubAlHizb.getRubAlHizb;
exports.getRubAlHizbByAyahId = require_getRubAlHizbByAyahId.getRubAlHizbByAyahId;
exports.getRubAlHizbMeta = require_getRubAlHizbMeta.getRubAlHizbMeta;
exports.getRubAlHizbMetaByAyahId = require_getRubAlHizbMetaByAyahId.getRubAlHizbMetaByAyahId;
exports.getRukuMeta = require_getRukuMeta.getRukuMeta;
exports.getSurahMeta = require_getSurahMeta.getSurahMeta;
exports.getSurahNames = require_getSurahNames.getSurahNames;
exports.isAyahJuzFirst = require_isAyahJuzFirst.isAyahJuzFirst;
exports.isAyahPageFirst = require_isAyahPageFirst.isAyahPageFirst;
exports.isSurahAyahJuzFirst = require_isSurahAyahJuzFirst.isSurahAyahJuzFirst;
exports.isSurahAyahPageFirst = require_isSurahAyahPageFirst.isSurahAyahPageFirst;
exports.isValidAyahId = require_typeGuards.isValidAyahId;
exports.isValidAyahNo = require_typeGuards.isValidAyahNo;
exports.isValidJuz = require_typeGuards.isValidJuz;
exports.isValidManzil = require_typeGuards.isValidManzil;
exports.isValidPage = require_typeGuards.isValidPage;
exports.isValidRuku = require_typeGuards.isValidRuku;
exports.isValidSurah = require_typeGuards.isValidSurah;
exports.isValidSurahAyah = require_typeGuards.isValidSurahAyah;
exports.languages = require_types.languages;
exports.nextAyah = require_nextAyah.nextAyah;
exports.prevAyah = require_prevAyah.prevAyah;
exports.quran = quran;
exports.string2NumberSplitter = require_ayahStringSplitter.string2NumberSplitter;
exports.string2NumberSplitterStrict = require_ayahStringSplitter.string2NumberSplitterStrict;
exports.surahNames = require_index$1.surahNames;
exports.surahNamesAz = require_surah_az.surahNamesAz;
exports.surahNamesBs = require_surah_bs.surahNamesBs;
exports.surahNamesEn = require_surah_en.surahNamesEn;
exports.surahNamesFr = require_surah_fr.surahNamesFr;
exports.surahNamesKk = require_surah_kk.surahNamesKk;
exports.surahNamesKy = require_surah_ky.surahNamesKy;
exports.surahNamesLt = require_surah_lt.surahNamesLt;
exports.surahNamesRu = require_surah_ru.surahNamesRu;
exports.surahNamesTg = require_surah_tg.surahNamesTg;
exports.surahNamesTr = require_surah_tr.surahNamesTr;
exports.surahNamesUz = require_surah_uz.surahNamesUz;
exports.surahStringParser = require_surahStringParser.surahStringParser;
