// quran-meta/lib_es/index.mjs
import { HafsLists } from "./lists/HafsLists.mjs";
import { QalunLists } from "./lists/QalunLists.mjs";
import { WarshLists } from "./lists/WarshLists.mjs";
import { checkValidAyahId, checkValidJuz, checkValidManzil, checkValidPage, checkValidRuku, checkValidSurah, checkValidSurahAyah } from "./validation.mjs";
import { getAyahCountInSurah } from "./getAyahCountInSurah.mjs";
import { isValidAyahId, isValidAyahNo, isValidJuz, isValidManzil, isValidPage, isValidRuku, isValidSurah, isValidSurahAyah } from "./typeGuards.mjs";
import { ayahStringSplitter, string2NumberSplitter, string2NumberSplitterStrict } from "./ayahStringSplitter.mjs";
import { findAyahIdBySurah } from "./findAyahIdBySurah.mjs";
import { findJuzByAyahId } from "./findJuzByAyahId.mjs";
import { findJuz } from "./findJuz.mjs";
import { findSurahAyahByAyahId } from "./findSurahAyahByAyahId.mjs";
import { findSurahByAyahId } from "./findSurahByAyahId.mjs";
import { findJuzAndShift, findJuzAndShiftByAyahId } from "./findJuzAndShift.mjs";
import { findJuzMetaBySurah } from "./findJuzMetaBySurah.mjs";
import { findManzilByAyahId } from "./findManzilByAyahId.mjs";
import { findManzil } from "./findManzil.mjs";
import { findPage } from "./findPage.mjs";
import { findPagebyAyahId } from "./findPagebyAyahId.mjs";
import { findRukuByAyahId } from "./findRukuByAyahId.mjs";
import { findRangeAroundAyah } from "./findRangeAroundAyah.mjs";
import { findRangeAroundSurahAyah } from "./findRangeAroundSurahAyah.mjs";
import { findRubAlHizbByAyahId } from "./findRubAlHizbByAyahId.mjs";
import { findRubAlHizb } from "./findRubAlHizb.mjs";
import { generatePartBlocks, getList, getListNormalised } from "./lists/getList.mjs";
import { getRubAlHizb } from "./getRubAlHizb.mjs";
import { getRubAlHizbByAyahId } from "./getRubAlHizbByAyahId.mjs";
import { getAyahMeta } from "./getAyahMeta.mjs";
import { getAyahMetasForSurah } from "./getAyahMetasForSurah.mjs";
import { getJuzMeta } from "./getJuzMeta.mjs";
import { getManzilMeta } from "./getManzilMeta.mjs";
import { getPageMeta } from "./getPageMeta.mjs";
import { getRubAlHizbMeta } from "./getRubAlHizbMeta.mjs";
import { getRubAlHizbMetaByAyahId } from "./getRubAlHizbMetaByAyahId.mjs";
import { getRukuMeta } from "./getRukuMeta.mjs";
import { getSurahMeta } from "./getSurahMeta.mjs";
import { prevAyah } from "./prevAyah.mjs";
import { nextAyah } from "./nextAyah.mjs";
import { isAyahJuzFirst } from "./isAyahJuzFirst.mjs";
import { isAyahPageFirst } from "./isAyahPageFirst.mjs";
import { isSurahAyahJuzFirst } from "./isSurahAyahJuzFirst.mjs";
import { isSurahAyahPageFirst } from "./isSurahAyahPageFirst.mjs";
import { surahStringParser } from "./surahStringParser.mjs";
import { QuranRiwaya } from "./QuranRiwaya.mjs";
import { getListOfRiwaya, getListsOfRiwaya } from "./lists/index.mjs";
import { surahNamesEn } from "./i18n/surah.en.mjs";
import { surahNamesAz } from "./i18n/surah.az.mjs";
import { surahNamesRu } from "./i18n/surah.ru.mjs";
import { surahNamesTr } from "./i18n/surah.tr.mjs";
import { surahNamesUz } from "./i18n/surah.uz.mjs";
import { surahNamesKk } from "./i18n/surah.kk.mjs";
import { surahNamesBs } from "./i18n/surah.bs.mjs";
import { surahNamesFr } from "./i18n/surah.fr.mjs";
import { surahNamesLt } from "./i18n/surah.lt.mjs";
import { surahNamesKy } from "./i18n/surah.ky.mjs";
import { surahNamesTg } from "./i18n/surah.tg.mjs";
import { getSurahNames } from "./i18n/getSurahNames.mjs";
import { languages } from "./i18n/types.mjs";
import { surahNames } from "./i18n/index.mjs";

//#region src/index.ts
/**
* Create a QuranRiwaya instance with Hafs riwaya (default)
*/
const createHafs = () => QuranRiwaya.create(HafsLists);
/**
* Create a QuranRiwaya instance with Qalun riwaya
*/
const createQalun = () => QuranRiwaya.create(QalunLists);
/**
* Create a QuranRiwaya instance with Warsh riwaya
*/
const createWarsh = () => QuranRiwaya.create(WarshLists);
/**
* Pre-initialized QuranRiwaya instance for Hafs (for convenience)
*/
const quran = createHafs();

//#endregion
export { QuranRiwaya, ayahStringSplitter, checkValidAyahId, checkValidJuz, checkValidManzil, checkValidPage, checkValidRuku, checkValidSurah, checkValidSurahAyah, createHafs, createQalun, createWarsh, findAyahIdBySurah, findJuz, findJuzAndShift, findJuzAndShiftByAyahId, findJuzByAyahId, findJuzMetaBySurah, findManzil, findManzilByAyahId, findPage, findPagebyAyahId, findRangeAroundAyah, findRangeAroundSurahAyah, findRubAlHizb, findRubAlHizbByAyahId, findRukuByAyahId, findSurahAyahByAyahId, findSurahByAyahId, generatePartBlocks, getAyahCountInSurah, getAyahMeta, getAyahMetasForSurah, getJuzMeta, getList, getListNormalised, getListOfRiwaya, getListsOfRiwaya, getManzilMeta, getPageMeta, getRubAlHizb, getRubAlHizbByAyahId, getRubAlHizbMeta, getRubAlHizbMetaByAyahId, getRukuMeta, getSurahMeta, getSurahNames, isAyahJuzFirst, isAyahPageFirst, isSurahAyahJuzFirst, isSurahAyahPageFirst, isValidAyahId, isValidAyahNo, isValidJuz, isValidManzil, isValidPage, isValidRuku, isValidSurah, isValidSurahAyah, languages, nextAyah, prevAyah, quran, string2NumberSplitter, string2NumberSplitterStrict, surahNames, surahNamesAz, surahNamesBs, surahNamesEn, surahNamesFr, surahNamesKk, surahNamesKy, surahNamesLt, surahNamesRu, surahNamesTg, surahNamesTr, surahNamesUz, surahStringParser };
